File: MassForge/_archive/3_概念设计工作流/test_box.py
"""Test rhino MCP box creation - simple line by line"""
import json
import subprocess
import time

CMD = ["D:/claude code mode/Python312/Scripts/uvx.exe", "rhinomcp"]

proc = subprocess.Popen(
    CMD,
    stdin=subprocess.PIPE,
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE,
    text=True,
    bufsize=1
)

def send(msg):
    proc.stdin.write(json.dumps(msg) + "\n")
    proc.stdin.flush()

def recv():
    line = proc.stdout.readline()
    if line:
        try:
            return json.loads(line.strip())
        except:
            return {"raw": line.strip()}
    return None

# Initialize
send({"jsonrpc":"2.0","id":1,"method":"initialize","params":{"protocolVersion":"2024-11-05","client":{"name":"test","version":"1.0"},"capabilities":{}}})
r = recv()
print("INIT:", json.dumps(r, ensure_ascii=False)[:200] if r else "NO RESPONSE")

# Initialized notification
send({"jsonrpc":"2.0","method":"notifications/initialized","params":{}})
time.sleep(1)

# List tools
send({"jsonrpc":"2.0","id":2,"method":"tools/list","params":{}})
r = recv()
if r and "result" in r:
    tools = r["result"].get("tools", [])
    print(f"\nTools ({len(tools)}):")
    for t in tools[:15]:
        print(f"  - {t['name']}")
else:
    print(f"\nTools response: {json.dumps(r, ensure_ascii=False)[:500] if r else 'TIMEOUT'}")

# Create box
send({"jsonrpc":"2.0","id":3,"method":"tools/call","params":{"name":"create_object","arguments":{"type":"BOX","params":{"width":20,"length":10,"height":5},"translation":[0,0,0],"color":[0,150,230],"name":"测试Box"}}})
r = recv()
print(f"\nBOX: {json.dumps(r, ensure_ascii=False)[:500] if r else 'TIMEOUT'}")

time.sleep(1)

proc.terminate()
proc.wait()

File: MassForge/_archive/3_概念设计工作流/test_box.js
const { spawn } = require('child_process');

const proc = spawn('D:/claude code mode/Python312/Scripts/uvx.exe', ['rhinomcp'], {
  stdio: ['pipe', 'pipe', 'pipe'],
  encoding: 'utf-8'
});

let buf = '';
proc.stdout.on('data', (data) => {
  buf += data.toString();
  const lines = buf.split('\n');
  buf = lines.pop();
  for (const line of lines) {
    if (!line.trim()) continue;
    try {
      const msg = JSON.parse(line.trim());
      handleMessage(msg);
    } catch (e) {}
  }
});

proc.stderr.on('data', (data) => {
  // drain stderr silently
});

let id = 1;
function send(method, params = {}) {
  const msg = { jsonrpc: '2.0', id: id++, method, params };
  proc.stdin.write(JSON.stringify(msg) + '\n');
}

let pending = {};

function handleMessage(msg) {
  if (msg.id) {
    const handler = pending[msg.id];
    if (handler) {
      handler(msg);
      delete pending[msg.id];
    }
  }
}

// Wait for connection, then initialize
setTimeout(() => {
  // Step 1: Initialize
  send('initialize', {
    protocolVersion: '2024-11-05',
    client: { name: 'test', version: '1.0' },
    capabilities: {}
  });
  pending[1] = (resp) => {
    console.log('INIT:', JSON.stringify(resp).substring(0, 200));

    // Step 2: Send initialized notification
    proc.stdin.write(JSON.stringify({
      jsonrpc: '2.0',
      method: 'notifications/initialized',
      params: {}
    }) + '\n');

    setTimeout(() => {
      // Step 3: List tools
      send('tools/list', {});
      pending[2] = (resp2) => {
        const tools = resp2.result?.tools || [];
        console.log(`\nTools (${tools.length}):`);
        tools.slice(0, 10).forEach(t => console.log(`  - ${t.name}`));

        // Step 4: Create box
        send('tools/call', {
          name: 'create_object',
          arguments: {
            type: 'BOX',
            params: { width: 20, length: 10, height: 5 },
            translation: [0, 0, 0],
            color: [0, 150, 230],
            name: 'TestBox'
          }
        });
        pending[3] = (resp3) => {
          console.log('\nBOX:', JSON.stringify(resp3).substring(0, 300));

          // Step 5: Screenshot
          send('tools/call', {
            name: 'capture_viewport',
            arguments: { viewport: 'perspective', width: 1200, height: 800, zoom_to_fit: true }
          });
          pending[4] = (resp4) => {
            console.log('\nSCREENSHOT:', JSON.stringify(resp4).substring(0, 200));
            proc.kill();
            process.exit(0);
          };
        };
      };
    }, 1000);
  };
}, 2000);

setTimeout(() => {
  console.log('TIMEOUT');
  proc.kill();
  process.exit(1);
}, 15000);
